src/System.Net.Http/tests/FunctionalTests/HttpClientTest.cs

```diff
@@ -30,5 +32,115 @@ public void Timeout_SetTo60AndGetResponseFromServerWhichTakes40_Success()
+        /// DiagnosticSources, since the global logging mechanism makes them conflict inherently.
+        /// </remarks>
+        [Fact]
+        public void SendAsync_ExpectedDiagnosticSourceLogging()
+        {
+            bool requestLogged = false;
+            Guid requestGuid = Guid.Empty;
+            bool responseLogged = false;
+            Guid responseGuid = Guid.Empty;
+
+            var diagnosticListenerObserver = new FakeDiagnosticListenerObserver(
+                kvp =>
+                {
+                    if (kvp.Key.Equals("System.Net.Http.Request"))
+                    {
+                        Assert.NotNull(kvp.Value);
+                        GetPropertyValueFromAnonymousTypeInstance<HttpRequestMessage>(kvp.Value, "Request");
+                        requestGuid = GetPropertyValueFromAnonymousTypeInstance<Guid>(kvp.Value, "LoggingRequestId");
+
+                        requestLogged = true;
+                    }
+                    else if (kvp.Key.Equals("System.Net.Http.Response"))
+                    {
+                        Assert.NotNull(kvp.Value);
+
+                        GetPropertyValueFromAnonymousTypeInstance<HttpResponseMessage>(kvp.Value, "Response");
+                        responseGuid = GetPropertyValueFromAnonymousTypeInstance<Guid>(kvp.Value, "LoggingRequestId");
+
+                        responseLogged = true;
+                    }
+                });
+            diagnosticListenerObserver.Enable();
+
+            DiagnosticListener.AllListeners.Subscribe(diagnosticListenerObserver);
+
+            using (var client = new HttpClient())
+            {
+                var response = client.GetAsync(HttpTestServers.RemoteEchoServer).Result;
+            }
+
+            Assert.True(requestLogged, "Request was not logged.");
+            // Poll with a timeout since logging response is not synchronized with returning a response.
+            WaitForTrue(() => responseLogged, TimeSpan.FromSeconds(1), "Response was not logged within 1 second timeout.");
+            Assert.Equal(requestGuid, responseGuid);
+
+            // No way to unsubscribe, so turn it off so that it doesn't interfere with other tests.
+            diagnosticListenerObserver.Disable(); 
+        }
+
+        /// <remarks>
+        /// This test must be in the same test collection as any others testing HttpClient/WinHttpHandler
+        /// DiagnosticSources, since the global logging mechanism makes them conflict inherently.
+        /// </remarks>
+        [Fact]
+        public void SendAsync_ExpectedDiagnosticSourceNoLogging()
+        {
+            bool requestLogged = false;
+            bool responseLogged = false;
+
+            DiagnosticListener.AllListeners.Subscribe(new FakeDiagnosticListenerObserver(
+                kvp =>
+                {
+                    if (kvp.Key.Equals("System.Net.Http.Request"))
```

Comment:
All of these strings such as "System.Net.Http.Request" should be factored out of these sources and into a Common file where we can track/rename all of them.  Please put them into a src/Common/src/System/Net/Http/* file and include it in the various CSPROJs.